runtime/wasm/src/types.rs

```diff
@@ -321,6 +322,10 @@ pub struct RuntimeContext {
     pub random_seed: Vec<u8>,
     /// Whether the execution should not charge any costs.
     pub free_of_charge: bool,
+    /// In case this is a function call on by the account owner on their account, we provide
```

Comment:
Could we always pass `public_key` to avoid making it an `Option`? This will make code more consistent.

Reply:
It's not available for receipts, only for self-calls

Reply:
It looks more consistent to me to have the same context fields available for self and regular contract calls. In case of regular contract calls `public_key` has the same meaning:
"public key to identify which access key or the public key was used to sign the initial transaction"

Reply:
Then we'd have pass it into all receipts. It maybe okay, but it's additional traffic

Reply:
I suggest for now we prioritize consistency over efficiency. It is in line with how we are replacing all transactions with contract calls. We can do optimizations later.

Reply:
Makes sense